src/task/SubscribeAndReceiveSubscribeAckTask.ts

```diff
@@ -94,6 +99,42 @@ export default class SubscribeAndReceiveSubscribeAckTask extends BaseTask {
     this.context.videoStreamIndex.integrateSubscribeAckFrame(subscribeAckFrame);
   }
 
+  // Our backends currently expect the video subscriptions passed in subscribe to precisely
+  // line up with the media sections, with a zero for any video send or inactive section.
+  //
+  // Firefox occasionally tosses stopped transceivers at the end of the SDP without reason
+  // and in general we don't want to be at the mercy of SDP sections not being in the same
+  // order as `getTransceivers`, so we simply recalculate the array here to enforce that
+  // expected invarient.
+  private fixUpSubscriptionOrder(sdp: string, videoSubscriptions: number[]): number[] {
+    const subscriptionsWithoutZero = videoSubscriptions.filter((value: number) => value !== 0);
+    let subscriptionsWithoutZeroIndex = 0;
+
+    const directions = new DefaultSDP(sdp).videoSectionDirections();
+    const newSubscriptions: number[] = [];
+    for (const direction of directions) {
+      if (direction === 'recvonly') {
+        if (subscriptionsWithoutZeroIndex >= subscriptionsWithoutZero.length) {
+          this.context.logger.warn(
```

Comment:
Can this actually happen? If it is unexpected, should it be at least error log?

Reply:
With browsers, i don't want to ever trust that the SDP will look how I expect, so I'm just trying to be safe here. Its not a hard failures (in fact there probably wouldn't be any impact), so I think a warning is sufficient.